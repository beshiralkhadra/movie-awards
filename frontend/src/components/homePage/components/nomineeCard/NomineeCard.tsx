import React from "react";
import "./index.css";
import Card from "../../../../shareableComponents/card/Card";
import { DefaultNomineeCardTypes } from "../../../../util/SharedTypes";
import { formatDate } from "../../../../util/sharedFunctions";

type NomineeCardTypes = DefaultNomineeCardTypes & {
  onSelect: () => void;
  isSelected: boolean;
};
const NomineeCard: React.FC<NomineeCardTypes> = React.memo(
  ({
    name,
    description,
    votes,
    stars,
    image_url,
    release_date,
    onSelect,
    isSelected,
  }: NomineeCardTypes) => {
    const [isExpanded, setIsExpanded] = React.useState(false);
    React.useEffect(() => {
      console.log("onSelect or isSelected changed");
    }, [onSelect, isSelected]);

    const toggleExpanded = () => {
      setIsExpanded(!isExpanded);
    };

    // const handleClick = () => {
    //   onSelect();
    // };

    return (
      <Card isSelected={isSelected}>
        <img src={image_url} alt={name} className="movie-card_img" />
        <div className="card-info">
          <h3>{name}</h3>
          <p>
            {isExpanded ? description : `${description.substring(0, 100)}...`}
            <span
              onClick={toggleExpanded}
              style={{ color: "#C79F27", cursor: "pointer" }}
            >
              {isExpanded ? " Show less" : " Show more"}
            </span>
          </p>
          <p>Stars: {stars.join(", ")}</p>
          <p>Release Date: {formatDate(release_date.toString())}</p>
          <div className="nominee-card_footer-section">
            <p>Votes: {votes}</p>
            <button id="vote" onClick={onSelect}>
              Vote
            </button>
          </div>
        </div>
      </Card>
    );
  }
);

export default NomineeCard;
