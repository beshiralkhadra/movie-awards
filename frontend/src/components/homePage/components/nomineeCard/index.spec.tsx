import React from "react";
import { shallow } from "enzyme";
import NomineeCard from "./NomineeCard";

describe("NomineeCard", () => {
  const mockProps = [
    {
      name: "test",
      description: "test",
      votes: 1,
      stars: ["test"],
      image_url: "test",
      release_date: new Date(),
      onSelect: jest.fn(),
      isSelected: false,
    },
  ];
  it("should render correctly", () => {
    const wrapper = shallow(<NomineeCard {...mockProps[0]} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should be clickable", () => {
    const wrapper = shallow(<NomineeCard {...mockProps[0]} />);
    wrapper.find('[id="vote"]').simulate("click");
    expect(mockProps[0].onSelect).toHaveBeenCalled();
  });
});
