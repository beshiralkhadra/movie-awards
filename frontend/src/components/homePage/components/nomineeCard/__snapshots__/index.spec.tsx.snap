// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NomineeCard should render correctly 1`] = `ShallowWrapper {}`;
